server/dataset/cxg_dataset.py

```diff
@@ -187,11 +196,24 @@ def _validate_and_initialize(self):
         self.about = about
         self.cxg_version = cxg_version
         self.corpora_props = corpora_props
+        self._init_sparsity_and_format()
 
     @staticmethod
     def _open_array(uri, tiledb_ctx):
         return tiledb.open(uri, mode="r", ctx=tiledb_ctx)
 
+    def open_X_array(self, col=False):
+        """
+        A helper function to open the 1D X array in row or column orientation with
+        backwards compatibility for 2D arrays.
+        """
+        if self.is_1d and col:
+            return self.open_array("Xc")
+        elif self.is_1d and not col:
+            return self.open_array("Xr")
+        else:
+            return self.open_array("X")
```

Comment:
if not using CXG version for conditional, worth noting in a comment here that `X` is deprecated

Reply:
X no longer deprecated - it is for dense data. Added a comment!